lib/logger/include/logger.h

```diff
@@ -38,11 +46,8 @@ class Logger
 	};
 
 	// printf-like logging; either const or non-const
-	//TO-DO: This causes compile errors in VS2015, fix it
-	//const Logger& operator()(const char*, ...)
-	//    __attribute__((format(printf, 2, 3)));
-	//const Logger& operator()(const char*, ...) const
-	//    __attribute__((format(printf, 2, 3)));
+	PRINTF_FORMAT
+	PRINTF_FORMAT_CONST
```

Comment:
This way you entirely removed these functions in Visual Studio, so you shouldn't be able to compile again. Try doing it as I commented earlier - use `#define ` only for the single `__attribute` macro, then here, on lines 49 and 50, use it like this:

```
const Logger& operator()(const char*, ...) PRINTF_FORMAT;
```

Another issue with the current update is that you repeat the code twice and it'd be harder to change it in the future.